x/ibc/03-connection/keeper/keeper.go

```diff
@@ -20,14 +20,16 @@ import (
 // Keeper defines the IBC connection keeper
 type Keeper struct {
 	storeKey     sdk.StoreKey
-	cdc          *codec.Codec
+	aminoCdc     *codec.Codec         // amino codec. TODO: remove after clients have been migrated to proto
+	cdc          clientexported.Codec // hybrid codec
 	clientKeeper types.ClientKeeper
 }
 
 // NewKeeper creates a new IBC connection Keeper instance
-func NewKeeper(cdc *codec.Codec, key sdk.StoreKey, ck types.ClientKeeper) Keeper {
+func NewKeeper(aminoCdc *codec.Codec, cdc clientexported.Codec, key sdk.StoreKey, ck types.ClientKeeper) Keeper {
```

Comment:
Why do you need amino @fedekunze?

Reply:
to call the client state verification functions